src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -122,6 +123,248 @@ static void verifySortingKey(const KeyDescription & sorting_key)
+
+        if (replica_name.empty())
+            throw Exception(ErrorCodes::NO_REPLICA_NAME_GIVEN, "No replica name in config{}", verbose_help_message);
+
+        expand_macro(ast_zk_path, ast_replica_name);
+    }
+    else if (is_extended_storage_def
+        && (arg_cnt == 0
+            || !engine_args[arg_num]->as<ASTLiteral>()
+            || (arg_cnt == 1 && (getNamePart(engine_name) == "Graphite"))))
+    {
+        /// Try use default values if arguments are not specified.
+        /// Note: {uuid} macro works for ON CLUSTER queries when database engine is Atomic.
+        const auto & server_settings = context->getServerSettings();
+        zookeeper_path = server_settings.default_replica_path;
+        /// TODO maybe use hostname if {replica} is not defined?
+        replica_name = server_settings.default_replica_name;
+
+        /// Modify query, so default values will be written to metadata
+        assert(arg_num == 0);
+        ASTs old_args;
+        std::swap(engine_args, old_args);
+        auto path_arg = std::make_shared<ASTLiteral>(zookeeper_path);
+        auto name_arg = std::make_shared<ASTLiteral>(replica_name);
+        auto * ast_zk_path = path_arg.get();
+        auto * ast_replica_name = name_arg.get();
+
+        expand_macro(ast_zk_path, ast_replica_name);
+
+        engine_args.emplace_back(std::move(path_arg));
+        engine_args.emplace_back(std::move(name_arg));
+        std::move(std::begin(old_args), std::end(old_args), std::back_inserter(engine_args));
+    }
+    else
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Expected two string literal arguments: zookeeper_path and replica_name");
+}
+
+/// Extracts a zookeeper path from a specified CREATE TABLE query. Returns std::nullopt if fails.
+std::optional<String> extractZooKeeperPathFromReplicatedTableDef(const ASTCreateQuery & query, const ContextPtr & context)
+{
+    try
+    {
+        if (!query.storage || !query.storage->engine)
+            return {};
+
+        const String & engine_name = query.storage->engine->name;
+        if (!isReplicated(engine_name))
+            return {};
+
+        StorageID table_id{query.getDatabase(), query.getTable(), query.uuid};
+        ASTs engine_args;
+        if (query.storage->engine->arguments)
+            engine_args = query.storage->engine->arguments->children;
+        for (auto & engine_arg : engine_args)
+            engine_arg = engine_arg->clone();
+        LoadingStrictnessLevel mode = LoadingStrictnessLevel::CREATE;
+        String zookeeper_path;
+        String replica_name;
+        RenamingRestrictions renaming_restrictions;
+
+        extractZooKeeperPathAndReplicaNameFromEngineArgs(query, table_id, engine_name, engine_args, mode, context,
+                                                         zookeeper_path, replica_name, renaming_restrictions);
+
+        return zookeeper_path;
+    }
+    catch (...)
+    {
+        return {};
```

Comment:
Should we at least log the exception? Do we really need to catch all exceptions or may be one concrete ones to avoid obscuring bugs?

Reply:
done